br/pkg/storage/gcs.go

```diff
@@ -224,6 +224,26 @@ func (s *gcsStorage) Create(ctx context.Context, name string) (ExternalFileWrite
 	return newFlushStorageWriter(wc, &emptyFlusher{}, wc), nil
 }
 
+// Rename file name from oldFileName to newFileName.
+func (s *gcsStorage) Rename(ctx context.Context, oldFileName, newFileName string) error {
+	data, err := s.ReadFile(ctx, oldFileName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = s.WriteFile(ctx, newFileName, data)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return s.DeleteFile(ctx, oldFileName)
+}
+
+// AtomicWriteFile implements ExternalStorage interface.
+func (s *gcsStorage) AtomicWriteFile(ctx context.Context, name string, data []byte) error {
```

Comment:
Is there any case we need to depend on `WriteFile` instead of `AtomicWriteFile`? If not I think we can repalce WriteFile with AtomicWriteFile then

Reply:
i'm not sure too, but i think we don't expect `WriteFile` may left an empty file, we need it success or failed and don't affect origin file. may be i can change `WriteFile` to be atomic and run our tests.